src/frontend/src/optimizer/plan_node/logical_project_set.rs

```diff
@@ -350,3 +372,58 @@ impl ToStream for LogicalProjectSet {
         Ok(StreamProjectSet::new(new_logical).into())
     }
 }
+
+#[cfg(test)]
+mod test {
+    use risingwave_common::catalog::{Field, Schema};
+    use risingwave_common::types::DataType;
+
+    use super::*;
+    use crate::expr::{ExprImpl, InputRef, TableFunction};
+    use crate::optimizer::plan_node::LogicalValues;
+    use crate::optimizer::property::FunctionalDependency;
+    use crate::session::OptimizerContext;
+
+    #[tokio::test]
+    async fn fd_derivation_project_set() {
```

Comment:
Can add some comments to make the test more readable. 
e.g., 
```
input: [v1,v2,v3], FD: v2->v3
output: [projected_row_id, v3,v2,generate_series(v1,v2,v3)], FD: [projected_row_id,v2]->v3
```
Also for other nodes